core/dbt/events/functions.py

```diff
@@ -116,23 +115,35 @@ def stop_capture_stdout_logs() -> None:
 # returns a dictionary representation of the event fields.
 # the message may contain secrets which must be scrubbed at the usage site.
 def event_to_json(
-    event: BaseEvent,
+    event: Event,
 ) -> str:
     event_dict = event_to_dict(event)
-    raw_log_line = json.dumps(event_dict, sort_keys=True)
+    raw_log_line = json.dumps(event_dict)
     return raw_log_line
 
 
-def event_to_dict(event: BaseEvent) -> dict:
+def event_to_dict(event: Event) -> dict:
+    """This creates a dictionary using the betterproto to_dict, but removing
+    all of the extra oneof detail events. We need defaults for our json
+    logs, but can't have the multiple detail attributes"""
     event_dict = dict()
     try:
         # We could use to_json here, but it wouldn't sort the keys.
         # The 'to_json' method just does json.dumps on the dict anyway.
+        # Note: we need to have include_default_values set to True here because we
+        # need defaults in the json log output, but that includes a huge unnecessary
+        # pile of oneof field dictionaries, so we'll have to munge it.
         event_dict = event.to_dict(casing=betterproto.Casing.SNAKE, include_default_values=True)  # type: ignore
     except AttributeError as exc:
         event_type = type(event).__name__
-        raise Exception(f"type {event_type} is not serializable. {str(exc)}")
-    return event_dict
+        raise BaseException(f"type {event_type} is not serializable. {str(exc)}")
+
+    # Kludge warning: create a new dictionary with only the relevant keys. In the
```

Comment:
Can you create an issue to investigate this and link to the issue in the comment?  That way it's less likely to get lost and when we read this we can know where to find more context.

Reply:
Created a ticket and added a comment.